crates/database/src/trades.rs

```diff
@@ -35,27 +81,53 @@ SELECT
     o.owner,
     o.buy_token,
     o.sell_token,
-    settlement.tx_hash
+    settlement.tx_hash,
+    COALESCE(json_agg(
+            json_build_object(
+                'auction_id', fp.auction_id,
+                'order_uid', fp.order_uid,
+                'kind', fp.kind,
+                'surplus_factor', fp.surplus_factor,
+                'surplus_max_volume_factor', fp.surplus_max_volume_factor,
+                'volume_factor', fp.volume_factor,
+                'price_improvement_factor', fp.price_improvement_factor,
+                'price_improvement_max_volume_factor', fp.price_improvement_max_volume_factor
+            ) ORDER BY fp.application_order
+        ) FILTER (WHERE fp.auction_id IS NOT NULL), '[]') AS fee_policies,
+    oq.sell_amount AS quote_sell_amount,
+    oq.buy_amount AS quote_buy_amount,
+    (oq.gas_amount * oq.gas_price / oq.sell_token_price) AS quote_fee
 FROM trades t
 LEFT OUTER JOIN LATERAL (
-    SELECT tx_hash FROM settlements s
+    SELECT tx_hash, auction_id FROM settlements s
     WHERE s.block_number = t.block_number
     AND   s.log_index > t.log_index
     ORDER BY s.log_index ASC
     LIMIT 1
 ) AS settlement ON true
-JOIN orders o
-ON o.uid = t.order_uid"#;
+JOIN orders o ON o.uid = t.order_uid
+LEFT JOIN fee_policies fp ON fp.auction_id = settlement.auction_id AND fp.order_uid = t.order_uid
+LEFT JOIN order_quotes oq ON oq.order_uid = t.order_uid"#;
     const QUERY: &str = const_format::concatcp!(
+        "WITH combined AS (",
         COMMON_QUERY,
         " WHERE ($1 IS NULL OR o.owner = $1)",
         " AND ($2 IS NULL OR o.uid = $2)",
-        "UNION",
+        " GROUP BY t.block_number, t.log_index, t.order_uid, o.owner, o.buy_token, o.sell_token, \
+         settlement.tx_hash, oq.sell_amount, oq.buy_amount, oq.gas_amount, oq.gas_price, \
+         oq.sell_token_price ",
+        " UNION ALL ",
         COMMON_QUERY,
         " LEFT OUTER JOIN onchain_placed_orders onchain_o",
         " ON onchain_o.uid = t.order_uid",
         " WHERE onchain_o.sender = $1",
         " AND ($2 IS NULL OR o.uid = $2)",
+        " GROUP BY t.block_number, t.log_index, t.order_uid, o.owner, o.buy_token, o.sell_token, \
+         settlement.tx_hash, oq.sell_amount, oq.buy_amount, oq.gas_amount, oq.gas_price, \
+         oq.sell_token_price ",
+        ") ",
+        "SELECT DISTINCT ON (block_number, log_index, order_uid) * FROM combined ",
+        "ORDER BY block_number DESC"
```

Comment:
UNION ALL is here now because JSON used for fee policies doesn't support the equality operator in Postgres. SELECT DISTINCT ON compensates for this limitation.